shared/src/commonTest/kotlin/io/github/MockWithNestedMockWithUnitReturningFunctionTests.kt

```diff
@@ -0,0 +1,29 @@
+package io.github
+
+import io.mockative.MissingExpectationException
+import io.mockative.Mock
+import io.mockative.classOf
+import io.mockative.every
+import io.mockative.mock
+import io.mockative.verify
+import kotlin.test.Test
+import kotlin.test.assertFailsWith
+
+class MockWithNestedMockWithUnitReturningFunctionTests {
+    @Mock
+    val mockedClassWithProperty = mock(classOf<ClassWithProperty>())
+    val mockedClassUsedAsPropertyWithUnitReturningFunction = mock(classOf<ClassUsedAsPropertyWithUnitReturningFunction>())
+
+    @Test
+    fun stubs_units_by_default_works_for_recursive_mock() {
+        every { mockedClassWithProperty.propertyWithUnitReturningFunction }.returns(mockedClassUsedAsPropertyWithUnitReturningFunction)
+        mockedClassWithProperty.propertyWithUnitReturningFunction.doSomethingReturnUnit()
+        verify { mockedClassWithProperty.propertyWithUnitReturningFunction.doSomethingReturnUnit() }.wasInvoked(1)
+    }
+
+    @Test
+    fun stubs_units_by_default_doesnt_affect_function_not_returning_unit() {
+        every { mockedClassWithProperty.propertyWithUnitReturningFunction }.returns(mockedClassUsedAsPropertyWithUnitReturningFunction)
+        assertFailsWith<MissingExpectationException> { mockedClassWithProperty.propertyWithUnitReturningFunction.returnSomeString() }
+    }
+}
```

Comment:
Hi @riQQ

Good work on the PR
However, I dont see how these tests asses that the added functionality works. 
Could you clarify what your thoughts are?

Reply:
@kris098e you're right, the tests don't work, because the property can't be accessed without mocking it, which overrides the actual value set via the constructor in the generated code. I don't know how to test this 'feature' and that makes me wonder if this feature really does anything.

```kotlin
public fun mock(@Suppress("UNUSED_PARAMETER") type: KClass<ClassUsedAsPropertyWithUnitReturningFunction>):
    ClassUsedAsPropertyWithUnitReturningFunction =
    configure(ClassUsedAsPropertyWithUnitReturningFunctionMock(null)) { stubsUnitByDefault = true }
```

Reply:
Yes, at the moment what is suggested does not work in Mockative.

We did however discuss implementing this. It should be done in a way where getting the properties that are mockative mocks should be returned if there are no stubbing for the getting the property. This is for instance in alignment with MockK.

This would allow to both have a fall back and also the ability to stub the getter if the user wants

---
Therefore what you have implemented here is very important for the project, if someone finds the time to implement the above specified feature.

But for now, we should drop the tests

Reply:
@kris098e done, tests removed